website/src/content/development/testing.mdx

```diff
@@ -30,3 +32,500 @@ To test React components, we recommend to use one of the following approaches:
+                },
+              },
+              response: {
+                balance: 300,
+              },
+            },
+          ],
+        }
+      );
+      await rendered.findByText('Loading...');
+      await wait(() => {
+        expect(rendered.queryByText('Your balance is 300€')).toBeInTheDocument();
+      });
+    });
+  });
+  ```
+
+* Using feature toggles
+
+  ```jsx
+  import { renderApp } from '@commercetools-frontend/application-shell/test-utils';
+  import { useFeatureToggle } from '@flopflip/react-broadcast';
+
+  const Profile = props => {
+    const showAge = useFeatureToggle('experimentalAgeOnProfileFlag');
+    return (
+      <div>
+        {props.name}
+        {props.showAge && `(${props.age})`}
+      </div>
+    );
+  };
+
+  describe('Profile', () => {
+    const baseProps = { name: 'Penny', age: 32 };
+
+    it('should show no age when feature is toggled off', async () => {
+      const rendered = renderApp(<Profile {...baseProps} />, {
+        flags: { experimentalAgeOnProfileFlag: false },
+      });
+      await rendered.findByText('Penny');
+      await wait(() => {
+        expect(rendered.queryByText('32')).not.toBeInTheDocument();
+      });
+    });
+
+    it('should show age when feature toggle is on', () => {
+      const rendered = renderApp(<Profile {...baseProps} />, {
+        flags: { experimentalAgeOnProfileFlag: true },
+      });
+      await rendered.findByText('Penny (32)');
+    });
+  });
+  ```
+
+  * Using the router
+
+  ```jsx
+  import { Switch, Route, Redirect } from 'react-router';
+  import { renderApp } from '@commercetools-frontend/application-shell/test-utils';
+
+  const ProductTabs = () => (
+    <Switch>
+      <Route path="/products/:productId/general" render={() => 'General'} />
+      <Route path="/products/:productId/pricing" render={() => 'Pricing'} />
+      {/* Define a catch-all route */}
+      <Redirect from="/products/:productId" to="/products/:productId/general" />
+    </Switch>
+  );
+
+  describe('router', () => {
+    it('should redirect to "general" when no tab is given', async () => {
+      const rendered = renderApp(<ProductTabs />, {
+        route: '/products/party-parrot',
+      });
+      await rendered.findByText('General');
+    });
+    it('should render "general" when on general tab', async () => {
+      const rendered = renderApp(<ProductTabs />, {
+        route: '/products/party-parrot/general',
+      });
+      await rendered.findByText('General');
+    });
+    it('should render "pricing" when on pricing tab', async () => {
+      const rendered = renderApp(<ProductTabs />, {
+        route: '/products/party-parrot/pricing',
+      });
+      await rendered.findByText('Pricing');
+    });
+  });
+  ```
+
+## Testing permissions
+
+User permissions are bound to a `project` and can vary depending on the permissions assigned to the team where the user belongs to.
+
+<Info>
+
+See [User Permissions](/main-concepts/user-permissions).
```

Comment:
FYI: @tdeekens this would be the page to document the permissions. Ref #1264

Reply:
Yup. Will try to get to it tomorrow or next week during the travels.